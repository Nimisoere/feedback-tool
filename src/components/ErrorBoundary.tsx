import React from "react";
import ServerErrorDislay from "./ServerErrorDislay/ServerErrorDislay";

interface Props {
  children: React.ReactNode;
}

interface State {
  error?: Error;
}

class ErrorBoundary extends React.Component<Props, State> {
  public state: State = {
    error: undefined,
  };

  public componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    this.setState({ error: error });
  }

  public render() {
    if (this.state.error) {
      return <ServerErrorDislay />;
    }

    return this.props.children;
  }
}

export default ErrorBoundary