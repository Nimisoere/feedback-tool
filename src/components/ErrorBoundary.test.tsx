import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import ErrorBoundary from "./ErrorBoundary";

const Child = () => {
  throw new Error("Test")
}

describe("Feedback Graph", () => {
  it("should match snapshot", async () => {
    const { getByText, container } = render(
      <ErrorBoundary>
        <Child />
      </ErrorBoundary>
    );
    expect(screen.getByTestId('server-error')).toBeVisible();
  });
});
