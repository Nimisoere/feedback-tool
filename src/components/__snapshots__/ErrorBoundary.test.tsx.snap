// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Graph should match snapshot 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col items-stretch"
  >
    <header
      class="w-full sticky top-0 z-10 shadow bg-white sm:h-20 py-4 sm:py-0 h-auto flex items-center transition-all duration-500"
    >
      <div
        class="container mx-auto flex flex-wrap justify-between"
      >
        <div>
          <a
            class="font-bold text-2xl"
            href="/"
          >
            FEEDr.
          </a>
        </div>
        <nav
          class="gap-4 sm:w-auto w-full flex justify-between"
        >
          <a
            class="btn-plain"
            href="/"
          >
            Give feedback
          </a>
          <a
            class="btn-plain"
            href="/view-feedback"
          >
            View feedback
          </a>
        </nav>
      </div>
    </header>
    <main
      class="flex-grow"
    >
      <div
        class="flex flex-col items-center text-center container mx-auto py-24"
      >
        <h1
          class="font-semibold font-serif text-4xl text-neutral-400 mb-8"
        >
          500 | Oops!
        </h1>
        <p>
          Sorry ðŸ˜”, It's not you, it's us. Something's gone wrong on our end.
          <br />
          <a
            class="underline decoration-amber-800"
            href="/"
          >
            We should be back shortly
          </a>
        </p>
      </div>
    </main>
  </div>
</div>
`;
